clang/include/clang/Driver/ToolChain.h

```diff
@@ -306,6 +306,14 @@ class ToolChain {
       SmallVectorImpl<llvm::opt::Arg *> &AllocatedArgs,
       Action::OffloadKind DeviceOffloadKind) const;
 
+  /// Gather command line arguments for backend compilation call
+  void TranslateBackendTargetArgs(const llvm::opt::ArgList &Args,
```

Comment:
Is there a way to put all of the needed functionality into the `SYCLToolChain`'s implementation `ToolChain::TranslateTargetArgs()`? Otherwise, I'm a little bit concerned that these non-virtual methods would be propagated into the `ToolChain`-derived entities that have no SYCL specifics to them, especially in the presence of what looks like a generic method for the same purpose.

I know this comment is a bit vague, and I'll research further myself, too. Sharing this right away because there likely are some limitations I'm missing, or the actual ways to implement this that you'd know already/come up with faster.

Reply:
I understand what you are getting at.  I'll look into it closer

Reply:
I did some refactoring to move the translating to the SYCL tool space.

Reply:
As a leftover, these methods should be removed from `ToolChain`

Reply:
Done.